source/contracts/reporting/ReputationToken.sol

```diff
@@ -30,57 +30,27 @@ contract ReputationToken is DelegationTarget, Extractable, ITyped, Initializable
-        require(universe.isContainerForStakeToken(IStakeToken(msg.sender)));
-        return internalMigrateOut(_destination, msg.sender, _reporter, _attotokens, true);
-    }
-
-    function migrateOutDisputeBond(IReputationToken _destination, address _reporter, uint256 _attotokens) public onlyInGoodTimes afterInitialized returns (bool) {
-        require(universe.isContainerForDisputeBond(IDisputeBond(msg.sender)));
-        return internalMigrateOut(_destination, msg.sender, _reporter, _attotokens, true);
-    }
-
-    function migrateOut(IReputationToken _destination, address _reporter, uint256 _attotokens) public onlyInGoodTimes afterInitialized returns (bool) {
-        return internalMigrateOut(_destination, msg.sender, _reporter, _attotokens, false);
-    }
-
-    // AUDIT: check for reentrancy issues here, _destination will be called as contracts during validation
-    function internalMigrateOut(IReputationToken _destination, address _sender, address _reporter, uint256 _attotokens, bool _bonusIfInForkWindow) private onlyInGoodTimes returns (bool) {
+    function migrateOut(IReputationToken _destination, uint256 _attotokens) public onlyInGoodTimes afterInitialized returns (bool) {
         assertReputationTokenIsLegit(_destination);
-        if (_sender != _reporter) {
-            // Adjust token allowance here since we're bypassing the standard transferFrom method
-            allowed[_reporter][_sender] = allowed[_reporter][_sender].sub(_attotokens);
-        }
-        burn(_reporter, _attotokens);
-        _destination.migrateIn(_reporter, _attotokens, _bonusIfInForkWindow);
-        if (topMigrationDestination == address(0) || _destination.totalSupply() > topMigrationDestination.totalSupply()) {
-            topMigrationDestination = _destination;
-        }
+        burn(msg.sender, _attotokens);
+        _destination.migrateIn(msg.sender, _attotokens);
         return true;
     }
 
-    function migrateIn(address _reporter, uint256 _attotokens, bool _bonusIfInForkWindow) public onlyInGoodTimes afterInitialized returns (bool) {
+    function migrateIn(address _reporter, uint256 _attotokens) public onlyInGoodTimes afterInitialized returns (bool) {
+        IUniverse _parentUniverse = universe.getParentUniverse();
         require(ReputationToken(msg.sender) == universe.getParentUniverse().getReputationToken());
         mint(_reporter, _attotokens);
-        // Only count tokens migrated toward the available to be matched in other universes. The bonus should not be added
-        universe.increaseRepAvailableForExtraBondPayouts(_attotokens);
-        if (eligibleForForkBonus(_bonusIfInForkWindow)) {
+        // Award a bonus if migration is done before the fork has resolved and check if the fork can be resolved early
+        if (!_parentUniverse.getForkingMarket().isFinalized()) {
```

Comment:
It is initialized to 0, but since we're calling from the supposed child universe it would be impossible for it to legitimately have no forking market.

Reply:
Gotcha, no test tube universes, all have a parent.